### PR TITLE
Refactor of the events system

### PR DESCRIPTION
**Pull request by OSPFNeighbour**
Needs #221 and #222

### CODE REVIEW

**Comment by OSPFNeighbour** [comment]
screenshot 2018-03-08 17 38 19
screenshot 2018-03-08 17 38 06

File: client/src/components/Container/Holder.js [hunk 1]
```diff
@@ -3,12 +3,17 @@ import React from 'react';
 import { View } from 'react-native';
 import styles from './styles';
 
-const Container = ({ children, margin, marginBot, transparent }) => {
+const Container = ({ children, margin, wide, marginBot, transparent }) => {
```
Comment (sjmcculloch) [inline client/src/components/Container/Holder.js]: Nice - exactly what it's meant to be used for

File: client/src/components/List/ListItem.js [hunk 2]
```diff
@@ -5,12 +5,14 @@ import Icon from 'react-native-vector-icons/FontAwesome';
 
 import styles from './styles';
 
-const ListItem = ({ title, subtitle, icon, onPress }) => (
+const ListItem = ({ title, supertitle, subtitle, icon, onPress }) => (
   <TouchableOpacity onPress={() => onPress()}>
     <View style={styles.container}>
       <View style={styles.textContainer}>
-        {subtitle && <Text style={styles.subtitleText}>{subtitle}</Text>}
+        {supertitle && <Text style={styles.subtitleText}>{supertitle}</Text>}
```
Comment (sjmcculloch) [inline client/src/components/List/ListItem.js]: Thank you for correcting this - makes much more sense now.

File: client/src/components/List/ListItemWide.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { TouchableOpacity, View, Text } from 'react-native';
+import PropTypes from 'prop-types';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import styles from './styles';
+
+const ListItemWide = ({ title, supertitle, subtitle, icon, onPress }) => (
```
Comment (sjmcculloch) [inline client/src/components/List/ListItemWide.js]: Approved but could this be the regular ListItem with a prop to make it wide?

File: client/src/screens/events/EventUsers.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+import { FlatList, Text, View } from 'react-native';
+
+import EVENT_QUERY from '../../graphql/event.query';
+
+import { Center, Container, Holder } from '../../components/Container';
+import { ListItem } from '../../components/List';
+import { Progress } from '../../components/Progress';
+import { Segment } from '../../components/Segment';
+
+class EventUsers extends Component {
```
Comment (sjmcculloch) [inline client/src/screens/events/EventUsers.js]: I like it :)

File: client/src/screens/events/StackEvents.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { StackNavigator } from 'react-navigation';
+
+import Events from './Events';
+import Event from './Detail';
+import EventResponseEdit from './ResponseEdit';
+import EventUsers from './EventUsers';
+import NavOptions from '../../config/NavOptions';
+
+const StackEvents = StackNavigator(
```
Comment (sjmcculloch) [inline client/src/screens/events/StackEvents.js]: Nice

**Comment by OSPFNeighbour** [comment]
screenshot 2018-03-10 22 53 05

who learnt CSS......... Tim learnt CSS

File: client/src/components/Container/styles.js [hunk 6]
```diff
@@ -6,7 +6,6 @@ const styles = StyleSheet.create({
     flex: 1,
     flexDirection: 'column',
     justifyContent: 'flex-start',
-    alignItems: 'center',
```
Comment (sdunster) [inline client/src/components/Container/styles.js]: did this affect anything else?

File: client/src/components/List/ListItem.js [hunk 7]
```diff
@@ -5,12 +5,16 @@ import Icon from 'react-native-vector-icons/FontAwesome';
 
 import styles from './styles';
 
-const ListItem = ({ title, subtitle, icon, onPress }) => (
+const ListItem = ({ title, bold, wide, supertitle, subtitle, icon, onPress }) => (
   <TouchableOpacity onPress={() => onPress()}>
-    <View style={styles.container}>
+    <View style={wide ? styles.containerWide : styles.container}>
       <View style={styles.textContainer}>
+        {supertitle && <Text style={styles.subtitleText}>{supertitle}</Text>}
```
Comment (sdunster) [inline client/src/components/List/ListItem.js]: style uses `subtitleText`. intentional?

File: client/src/components/List/styles.js [hunk 8]
```diff
@@ -2,25 +2,45 @@ import { extendAppStyleSheet } from '../../screens/style-sheet';
...
+    marginRight: 0,
     marginTop: 0,
     marginBottom: 6,
   },
   textContainer: {
-    flex: 1,
     flexDirection: 'column',
+    flexShrink: 1,
+  },
+  supertitle: {
```
Comment (sdunster) [inline client/src/components/List/styles.js]: name this `supertitleText` to be consistent?

File: client/src/screens/events/Detail.js [hunk 9]
```diff
@@ -123,30 +72,37 @@ class EventHeader extends React.Component {
...
-  }
+  };
 
   focusMap(animated) {
     this.map.fitToElements(animated);
   }
 
   render() {
-    const { name, details, responses, eventLocations } = this.props.event;
+    const { name, details, permalink, responses, eventLocations } = this.props.event; // eslint-disable-line
```
Comment (sdunster) [inline client/src/screens/events/Detail.js]: please specify the rule:
```
// eslint-disable-line rule/my-rule
```

File: client/src/components/Container/styles.js (hunk 6 above)
Comment (OSPFNeighbour) [reply client/src/components/Container/styles.js]: not that we can find

File: client/src/components/List/styles.js (hunk 8 above)
Comment (OSPFNeighbour) [reply client/src/components/List/styles.js]: nice catch

File: client/src/components/List/ListItem.js (hunk 7 above)
Comment (OSPFNeighbour) [reply client/src/components/List/ListItem.js]: nice catch

**Pull request merged**